### PR TITLE
[3377] Add existing user to lead school & provider

### PR DESCRIPTION
**Pull request by richardpattinson**
### Context

Add dropdowns to add providers and lead schools to a user in the system admin section of the application.

### Guidance to review

Go to /system-admin/users
click on a user
test the 'add a provider' and 'add a lead school' link

### Changes proposed in this pull request

New dropdown select screen added to create associations between users and lead schools/ providers

### CODE REVIEW

File: spec/support/page_objects/users/show.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      element :link, ".add-lead-school-to-user"
+    end
+
+    class Show < PageObjects::Base
+      set_url "/system-admin/users{/id}"
+      element :add_provider, "a", text: "Add provider"
+      element :add_lead_school, "a", text: "Add lead school"
+      elements :providers, ".provider-row dd a"
+      elements :lead_schools, ".lead-school-row dd a"
+      # sections :add_provider, AddProviderLink, ".provider-row"
```
Comment (gpeng) [inline spec/support/page_objects/users/show.rb:19]: No comment 😆

**Review by gpeng — CHANGES_REQUESTED** [review]
Looks good. Can you fix the formatting on the 'Add lead school' and 'Add provider' pages.

Screenshot 2022-01-26 at 10 47 47

The spacing isn't correct.

We should also make those an autocomplete but I'll card that up and we can do it 'in the future ™️' as it's not a priority for this sprint.

File: spec/support/page_objects/users/show.rb (hunk 1 above)
Comment (richardpattinson) [reply spec/support/page_objects/users/show.rb:19]: When you say fix the spacing, are we talking about leaving a space between the dropdown and the continue button, does the alignment of anything need to change?

Comment (richardpattinson) [reply spec/support/page_objects/users/show.rb]: ok ive changed the spacing and removed the comments, let me know if there are any other changes needed =)

File: db/schema.rb [hunk 2]
```diff
@@ -178,9 +178,6 @@
     t.index ["uuid"], name: "index_courses_on_uuid", unique: true
   end
 
-  create_table "data_migrations", primary_key: "version", id: :string, force: :cascade do |t|
```
Comment (gpeng) [inline db/schema.rb:181]: We decided we were going to keep this in main. It probably got removed as you don't have it on your local database.

File: spec/features/system_admin/users/add_a_lead_school_to_user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+feature "Creating a new lead school for a user" do
```
Comment (gpeng) [inline spec/features/system_admin/users/add_a_lead_school_to_user_spec.rb:5]: Sorry... really nit picky we've used all lower case for describe|context|feature|scenario|it strings in all of the other specs.

**Review by gpeng — CHANGES_REQUESTED** [review]

File: spec/features/system_admin/users/add_a_lead_school_to_user_spec.rb (hunk 3 above)
Comment (richardpattinson) [reply spec/features/system_admin/users/add_a_lead_school_to_user_spec.rb]: Fair enough, I've made the changes =)

**Review by gpeng — APPROVED** [review]

**Pull request merged**